client/question/services/FeedbackGeneratorService.js

```diff
@@ -263,33 +328,70 @@ tie.factory('FeedbackGeneratorService', [
      * @param {string|null} outputFunctionName If needed, is string of the
      *    name for the function needed to process/format the user's output.
      * @param {CorrectnessTest} TestCase
+     * @param {number} testCaseIndex
      * @param {*} observedOutput Actual output for running user's code.
+     * @param {string} testSuiteId
      * @returns {Feedback}
      * @private
      */
     var _getCorrectnessTestFeedback = function(
-      outputFunctionName, testCase, observedOutput) {
+        outputFunctionName, testCase, testCaseIndex, observedOutput,
+        testSuiteId) {
+      var testCaseStateKey = testSuiteId + '-' + testCaseIndex;
```

Comment:
This key just encodes a description of the test case, not its state, so you can call it testCaseKey.

Reply:
done